### DISCUSSION TITLE
O Bootstrap está ficando "ultrapassado"?

### DISCUSSION THREAD

**Discussion by peidrao** [frontendbr/forum#1737] [Discussão]
Acredito que o bootstrap tenha sido de grande ajuda para muitos da área do frontend, pois podemos criar uma base bem profissional usando apenas recortes que o próprio bootstrap oferece. _Ora, nunca foi tão fácil deixar as coisas, bem estruturadas e responsivas_. Contudo **em minha visão** (e isso é um olhar que tenho, então não tragam a cruz para me crucificar), o mesmo está ficando ultrapassado, se formos comparar com tantas alternativas que temos hoje em dia. 
Acredito que esse termo (ultrapassado) é um tanto errôneo, contudo é o que vem em meu pensamento.

**Comment by theandersonn** [comment]
**O framework em si não**. A atualizações que a versão 4 são até bem legais.

Agora, a "pratica/ação" de injetá-lo integralmente num projeto sem saber o que tem lá dentro, **está sim**!

Sou a favor de recorta-lo de extrair os componentes que realmente formos utilizar na aplicação, exemplo grid. Fiz isso [aqui](https://github.com/theandersonn/bootstrap-4-grid).

Outro fator que devemos levar em consideração é a evolução na diagramação, recursos como "flexbox" e "grid layout" podem tranquilamente substituir um grid ou outra forma de layoutar um site.

**Comment by tarciozemel** [comment]
Concordo plenamente com o @theandersonn.

Por curiosidade, traduzi um artigo com alguns comparativos práticos interessantes Bootstrap x CSS Grid: https://desenvolvimentoparaweb.com/css/css-grid-melhor-que-bootstrap/

Claro, também devemos levar em consideração que ele é muito popular pelas webs à fora, então quem trabalha pegando/manutenindo vários projetos (produtoras, agências, "software houses" e afins) deve ter ciência que ainda vai se deparar com o Boostrap por um booom tempo.

PS: às vezes, penso que os componentes do Bootstrap (incluindo a parte JS) são os responsáveis por seu sucesso/popularidade e o pessoal usa a grid meio que de lambuja. "Ah, já tá aqui mesmo..."

**Comment by peidrao** [comment]
Muito interessante o comentário como também seu texto. @tarciozemel você vê no mercado algum concorrente ao Bootstrap?

**Comment by ninetails** [comment]
Creio que a evolução é a mesma do que aconteceu ao **jQuery**. Vale a pena conhecer até hoje, porém com a atualização do ambiente web (adoção de browsers mais novos com _rolling release_, ou seja, atualização automática) torna-se algo menos atrativo a se desenvolver, pois alternativas open source com código sem compatibilidade para navegadores antigos começam a surgir, também paralelamente melhora-se a própria API dos navegadores, que é feita por comunidade (ou comitês envolvendo mais de uma empresa desenvolvedora) para adoção de padrões (**standards**), como o caso já citado do _flexbox_ e _grid layout_.

Ainda sim há alternativas a adoção do Bootstrap saindo da diagramação colunada e indo para componentes gráficos utilizando _Design Systems_ de empresas, sendo o maior exemplo o **Material UI** do Google. E como frameworks para frontend atualmente também se arquitetam por componentização, é possível ver bibliotecas prontas contendo componentes complexos para o framework favorito.

Tem também a vertente dos web components & Polymer com vários componentes prontos...

> Muito interessante o comentário como também seu texto. @tarciozemel você vê no mercado algum concorrente ao Bootstrap?

Dois frameworks de CSS que estão crescendo ultimamente seriam:
- [Tailwind](https://tailwindcss.com/)
- [Bulma](https://bulma.io/)

**Comment by peidrao** [comment]
Comentário muito rico @ninetails muito obrigado!

**Comment by tarciozemel** [comment]
@peidrao Pessoalmente, sempre preferi Foundation a Bootstrap. =P

**Comment by italosilva01-zz** [comment]
Então, ainda vale apena estudar bootstrap hj em dia ?

**Comment by peidrao** [comment]
@tarciozemel qual as vantagens do Foundation em relação ao Bootstrap?

**Comment by peidrao** [comment]
> Então, ainda vale apena estudar bootstrap hj em dia ?

Eu acredito que vale sim (pelo menos em minha opinião). Ganhei muito conhecimento usando Bootstrap, e tem a vantagem de sua documentação ser bem prática.

**Comment by fredericovicente** [comment]
Olá a todos..

Tenho uma dúvida e irei aproveitar este tópico para vos perguntar..

A um mês comecei a estudar e aprender mais sobre html5, Css3 e JS. Porém até o momento estou fazendo alguns cursos de em vídeo online, e a frenquentar bootcamps.

Pergunto a você;

Como não tenho experiência alguma e comecei estudando e digitando os códigos, vocês acham que a utilização do Bootstrap para meu aprendizado seria válido, ou só para quem já tem algum nível de experiência e utiliza o mesmo para ter mais velocidade em escrever os códigos e entrega de projetos?

Obrigado a todos.

**Comment by theandersonn** [comment]
@fredericovicente é importante que você entenda a forma como o bootstrap funciona e como ele lhe ajuda. 

_Simplificando tem que conseguir solucionar problemas com e sem bootstrap._

Por exemplo, o **grid**.

O que o grid, faz?
Em que circunstâncias preciso aplicar um grid?
Terei suporte em que browsers?

Uma observação é que o bootstrap < 4 usa float para a diagramção e > 4 aplica flexbox. 

Esse é apenas um exemplo, o bootstrap é grande e oferece muitas soluções, se não entendermos como utiliza-las passaremos a brigar com a ferramenta ao invés de utilizar com sabedoria as suas soluções.

**Comment by fredericovicente** [comment]
@theandersonn

Muito obrigado pela informação e dicas.

Vou dar um esforço para aprender ao máximo a usar o bootstrap e também focar em resolução de problemas.

Valeu...
